src/nested_pandas/nestedframe/core.py

```diff
@@ -153,14 +160,14 @@ def map_true_index(index):
         # Recover some truncation formatting, limited to head truncation
         if pd.get_option("display.max_rows") is None:
             return repr.to_html(max_rows=0)
-        elif repr.data.shape[0] > pd.get_option("display.max_rows"):
+        elif df_shape[0] > pd.get_option("display.max_rows"):
```

Comment:
I think I understand this change.  Is the idea to avoid creating a monster HTML representation *first* and then clipping it?  Or is it about having an accurate count of the original shape (`df_shape`) for the test?  (Or both, actually, is what it looks like.  I believe I understand now.)

Reply:
Right, the change to df_shape here is basically what you describe, but more specifically because repr.data may now be truncated, we want to use the original shape information to determine the truncation setting to display (basically to preserve the "..." that you see at the end of a truncated dataframe).